pkgs/by-name/si/sieve/package.nix

```
@@ -0,0 +1,109 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  electron,
+  copyDesktopItems,
+  makeDesktopItem,
+  stdenv,
+  xcbuild,
+}:
+
+buildNpmPackage rec {
+  pname = "sieve";
+  version = "0.6.1-unstable-2024-07-25";
+
+  src = fetchFromGitHub {
+    owner = "thsmi";
+    repo = pname;
+    rev = "5879679ed8d16a34af760ee56bfec16a1a322b4e";
+    hash = "sha256-wl6dwKoGan+DrpXk2p1fD/QN/C2qT4h/g3N73gF8sOI=";
+  };
+
+  npmDepsHash = "sha256-a2I9csxFZJekG1uCOHqdRaLLi5v/BLTz4SU+uBd855A=";
+
+  ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+
+  npmBuildScript = "gulp";
+  npmBuildFlags = "app:package";
+
+  nativeBuildInputs =
+    lib.optionals stdenv.isLinux [ copyDesktopItems ]
+    ++ lib.optionals stdenv.isDarwin [ xcbuild ];
+
+  installPhase = ''
+    runHook preInstall
+
+    build_dir=build/electron/resources
+
+    ${lib.optionalString stdenv.isLinux ''
+      mkdir -p $out/{share,bin}
+      cp -r $build_dir $out/share/sieve
+
+      # copy icon
+      icon_dir=$out/share/icons/hicolor/64x64/apps
+      mkdir -p $icon_dir
+      cp $build_dir/libs/icons/linux.png $icon_dir/sieve.png
```

The `icon_dir` variable seems redundant as it's only used once. We can also use `install` to create the destination and copy the file in one line:

```suggestion
      install -D $build_dir/libs/icons/linux.png $out/share/icons/hicolor/64x64/apps/sieve.png
```